lib/HLSL/DxilNoops.cpp

```diff
@@ -582,6 +583,93 @@ Pass *llvm::createDxilRewriteOutputArgDebugInfoPass() {
 
 INITIALIZE_PASS(DxilRewriteOutputArgDebugInfo, "dxil-rewrite-output-arg-debug-info", "Dxil Rewrite Output Arg Debug Info", false, false)
 
+//==========================================================
+// Reader pass
+//
+
+namespace {
+
+class DxilReinsertNops : public ModulePass {
+public:
+  static char ID;
+
+  DxilReinsertNops() : ModulePass(ID) {
+    initializeDxilReinsertNopsPass(*PassRegistry::getPassRegistry());
+  }
+
+  // In various linking scenarios, the dx.nothing.a variable might be prefixed
+  // and/or suffixed with something:
+  //
+  //   <library_name>.dx.nothing.a.<another_thing>
+  //
+  // This routine looks for the "dx.nothing.a" string inside of it, and as long
+  // as it's used in the expected way:
+  // 
+  // %0 = load i32, i32* getelementptr inbounds ([1 x i32], [1 x i32]* @dx.nothing.a, i32 0, i32 0)
+  // 
+  // ...it is deemed a valid nop.
+  //
+  static bool IsLegalNothingVarName(StringRef Name) {
```

Comment:
I think we have a couple of other special globals like this. 

1. Lifetime markers prior to 6.6
2. The dx.break global

Do they need this special handling for linking as well?

Pretty sure there is one for lifetime markers prior to 6.6,

Reply:
Yeah we might need similar handling for `@dx.break`, but I'm not sure what lifetime-marker needs. I think it's just a `store undef`.

Reply:
I just noticed another global `dx.preserve.value` that looks related to debugging

https://github.com/microsoft/DirectXShaderCompiler/blob/main/lib/HLSL/DxilNoops.cpp#L74

Should we handle that here as well?

Reply:
Those were for an experimental way to do Od and also preserve dbg.value order. It turned out to produce way too many values and we have better solutions now anyway, so they should be removed. I just haven't had the chance to yet.